config/templates/cloud-config-controller

```diff
@@ -380,14 +327,241 @@ write_files:
       exec nsenter -m -u -i -n -p -t 1 -- /usr/bin/rkt "$@"
 
 {{ if .UseCalico }}
-  - path: /opt/bin/install-calico-system
-    permissions: 0700
+  - path: /srv/kubernetes/manifests/calico.yaml
+    content: |
+      kind: ConfigMap
+      apiVersion: v1
+      metadata:
+        name: calico-config 
+        namespace: kube-system
+      data:
+        etcd_endpoints: "{{ .EtcdEndpoints }}"
+        cni_network_config: |-
+          {
+              "name": "calico",
+              "type": "flannel",
+              "delegate": {
+                  "type": "calico",
+                  "etcd_endpoints": "__ETCD_ENDPOINTS__",
+                  "etcd_key_file": "__ETCD_KEY_FILE__",
+                  "etcd_cert_file": "__ETCD_CERT_FILE__",
+                  "etcd_ca_cert_file": "__ETCD_CA_CERT_FILE__",
+                  "log_level": "info",
+                  "policy": {
+                      "type": "k8s",
+                      "k8s_api_root": "https://__KUBERNETES_SERVICE_HOST__:__KUBERNETES_SERVICE_PORT__",
+                      "k8s_auth_token": "__SERVICEACCOUNT_TOKEN__"
+                  },
+                  "kubernetes": {
+                      "kubeconfig": "__KUBECONFIG_FILEPATH__"
+                  }
+              }
+          }
+
+        etcd_ca: "/calico-secrets/etcd-ca"
+        etcd_cert: "/calico-secrets/etcd-cert"
+        etcd_key: "/calico-secrets/etcd-key"
+
+      ---
+
+      apiVersion: v1
+      kind: Secret
+      type: Opaque
+      metadata:
+        name: calico-etcd-secrets
+        namespace: kube-system
+      data:
+        etcd-key: "$ETCDKEY"
+        etcd-cert: "$ETCDCERT"
+        etcd-ca: "$ETCDCA"
+
+      ---
+
+      kind: DaemonSet
+      apiVersion: extensions/v1beta1
+      metadata:
+        name: calico-node
+        namespace: kube-system
+        labels:
+          k8s-app: calico-node
+      spec:
+        selector:
+          matchLabels:
+            k8s-app: calico-node
+        template:
+          metadata:
+            labels:
+              k8s-app: calico-node
+            annotations:
+              scheduler.alpha.kubernetes.io/critical-pod: ''
+              scheduler.alpha.kubernetes.io/tolerations: |
+                [{"key": "dedicated", "value": "master", "effect": "NoSchedule" },
```

Comment:
Sorry for back and forth but would you please make key be `node.alpha.kubernetes.io/role` rather than `dedicated` according to https://github.com/kubernetes/kubernetes/issues/36272?